libc/__cxa_atexit.c

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "../enclave/core/atexit.h"
```

Comment:
Do not include headers by relative traversal of the file hierarchy.

For libc implementations that need a specific OE runtime implementation, the corelibc headers already expose `oe_atexit()` which is a thin wrapper around `__cxa_atexit(),` which should have just been redefined to tuse `oe_cxa_atexit()`. If the `__cxa_atexit` symbol needs to be defined in libc outside of its use in `oe_atexit()`, then this implementation can just use the already exposed `oe_atexit()` with no loss of semantics.

If you still really want to expose an additional separate `oe_cxa_atexit()`, it should be broken out into a separate header under `include/openenclave/corelibc/bits` or `include/openenclave/internal`.

Reply:
Thank you for reviewing.

Note that **__cxa_atexit** takes an argument which is eventually passed to **func**, but **atexit** does not. So we definitely need **oe_cxa_atexit**.

I create **<openenclave/internal/cxa_atexit.h>** to house the function declaration.

Reply:
Changes look good; clean up the PR with the squash and we can merge it.